.github/workflows/pytest-gpu.yml

```diff
@@ -81,6 +81,8 @@ jobs:
               git fetch
               git checkout ${{ steps.extract_branch.outputs.branch }}
           fi
+          # Ensure any new requirements are installed
```

Comment:
This is a bit of a hack as we actually need to tweak the image used for testing. It is good for us to think how we treat these cases as a new image will be needed every time we update reqs.

Reply:
It's not hack, it's just ensuring the correct requirements are installed if they get changed. Saying it's a hack is a bit like saying 'git pull' is a hack.

Reply:
At some point yes, but only when an update occurs that requires to modify something fundamental on the image.